server/store/user_store.go

```diff
@@ -67,6 +83,38 @@ func (s *pluginStore) StoreUser(user *User) error {
 		return err
 	}
 
+	var allUsers []*UserShort
```

Comment:
0/5 aggregating multiple keys should be all done in the `api` layer. The `store` layer should define the persistent structs, and provide (atomic/simple) operations on them, the lowest common denominator; abstracting away only things like key hashing, encryption, simple collision handling, etc.

See https://github.com/levb/mattermost-plugin-solar-lottery/blob/master/server/store/rotation_store.go#L11 for an example.